### PR TITLE
Revert "Work around https://hibernate.atlassian.net/browse/HHH-16832"

### PR DESCRIPTION
**Pull request by yrodiere**
This reverts commit 823591f1e1fab10ba4a24f4c1d719a84bdb426bd.

As suggested [here](https://github.com/quarkusio/quarkus/pull/34441#issuecomment-1615692835) after the upgrade to Hibernate ORM 6.2.6.Final.

### CODE REVIEW

**Comment by yrodiere** [comment]
Sorry, can't do right now, too many urgent things to do. I'll get back to it when possible.

**Comment by yrodiere** [comment]
So the fixed worked only partially, we still need to fix https://hibernate.atlassian.net/browse/HHH-17034

**Comment by yrodiere** [comment]
> Let's rebase this after #35690

Done, let's see.

**Comment by yrodiere** [comment]
It works now, but I'll need some approval before I can merge :) @Sanne?

**Review by Sanne — APPROVED** [review]

**Pull request merged**
